frame/contracts/src/gas.rs

```diff
@@ -250,7 +250,11 @@ pub fn refund_unused_gas<T: Trait>(
 pub fn approx_gas_for_balance<Balance>(gas_price: Balance, balance: Balance) -> Gas
 	where Balance: AtLeast32Bit
 {
-	(balance / gas_price).saturated_into::<Gas>()
+	if gas_price.is_zero() {
+		Zero::zero()
+	} else {
+		(balance / gas_price).saturated_into::<Gas>()
+	}
```

Comment:
alternatively you could write `balance.checked_div(gas_price).unwrap_or_else(|| Zero::zero()).saturating_into::<Gas>()`